[Bug]: Unable to sync garmin activities

### What happened?

Sync for wellness data works
Activities are not being synced.
Batch fails with following error:
 [2026-04-19T19:49:12.707Z] [ERROR] Error creating/updating exercise entry with snapshot: error: invalid input syntax for type integer: "143.65133333333333"

### Steps to reproduce

1. Set up garmin sync
2. Do a manual sync
3. Activity gets created, but contains no metrics.
4. No other activities are created.

### Expected behavior

Activities should be created.
The number should be converted into integer before importing into the database or the database type needs to be changed to float.

### How are you running SparkyFitness?

Docker Compose

### SparkyFitness version

0.15.5.8

### Which area is affected?

Server / API

### Relevant logs or screenshots

```shell
[2026-04-19T19:49:12.707Z] [ERROR] Error creating/updating exercise entry with snapshot: error: invalid input syntax for type integer: "143.65133333333333"
sparkyfitness-server-1    |     at /app/SparkyFitnessServer/node_modules/.pnpm/pg@8.20.0/node_modules/pg/lib/client.js:631:17
sparkyfitness-server-1    |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
sparkyfitness-server-1    |     at async _createExerciseEntryWithClient (/app/SparkyFitnessServer/models/exerciseEntry.ts:497:27)
sparkyfitness-server-1    |     at async Object.createExerciseEntry (/app/SparkyFitnessServer/models/exerciseEntry.ts:579:19)
sparkyfitness-server-1    |     at async processGarminSimpleActivity (/app/SparkyFitnessServer/services/garminService.ts:671:20)
sparkyfitness-server-1    |     at async processActivitiesAndWorkouts (/app/SparkyFitnessServer/services/garminService.ts:63:9)
sparkyfitness-server-1    |     at async Object.syncGarminData (/app/SparkyFitnessServer/services/garminService.ts:896:33)
sparkyfitness-server-1    |     at async <anonymous> (/app/SparkyFitnessServer/routes/garminRoutes.ts:423:20) {
sparkyfitness-server-1    |   length: 162,
sparkyfitness-server-1    |   severity: 'ERROR',
sparkyfitness-server-1    |   code: '22P02',
sparkyfitness-server-1    |   detail: undefined,
sparkyfitness-server-1    |   hint: undefined,
sparkyfitness-server-1    |   position: undefined,
sparkyfitness-server-1    |   internalPosition: undefined,
sparkyfitness-server-1    |   internalQuery: undefined,
sparkyfitness-server-1    |   where: "unnamed portal parameter $24 = '...'",
sparkyfitness-server-1    |   schema: undefined,
sparkyfitness-server-1    |   table: undefined,
sparkyfitness-server-1    |   column: undefined,
sparkyfitness-server-1    |   dataType: undefined,
sparkyfitness-server-1    |   constraint: undefined,
sparkyfitness-server-1    |   file: 'numutils.c',
sparkyfitness-server-1    |   line: '615',
sparkyfitness-server-1    |   routine: 'pg_strtoint32_safe'
sparkyfitness-server-1    | }
```

### Anything else?

_No response_

--- a/SparkyFitnessServer/services/garminService.ts
+++ b/SparkyFitnessServer/services/garminService.ts
@@ -515,7 +515,9 @@ async function processGarminWorkoutSession(
         source_id: activity.activityId
           ? `${activity.activityId}_${exerciseSortOrder}`
           : null,
-        steps: activity.steps || activity.totalSteps || activity.stepCount || 0,
+        steps: Math.round(
+          activity.steps || activity.totalSteps || activity.stepCount || 0
+        ),
       };
       await exerciseEntryRepository.createExerciseEntry(
         userId,
@@ -659,14 +661,20 @@ async function processGarminSimpleActivity(
   const exerciseEntryData = {
     exercise_id: exercise.id,
     duration_minutes: activity.duration || 0,
-    calories_burned: activity.active_calories || 0,
+    calories_burned: Math.round(activity.active_calories || 0),
     entry_date: entryDate,
     notes: `Garmin Activity: ${activity.activityName} (${activity.activityType?.typeKey})`,
     distance: activity.distance,
     avg_heart_rate:
-      activity.averageHR || activity.averageHeartRateInBeatsPerMinute || null,
+      activity.averageHR || activity.averageHeartRateInBeatsPerMinute
+        ? Math.round(
+            activity.averageHR || activity.averageHeartRateInBeatsPerMinute
+          )
+        : null,
     source_id: activity.activityId?.toString() ?? null,
-    steps: activity.steps || activity.totalSteps || activity.stepCount || 0,
+    steps: Math.round(
+      activity.steps || activity.totalSteps || activity.stepCount || 0
+    ),
   };
   const newEntry = await exerciseEntryRepository.createExerciseEntry(
     userId,

